commons/commons-bio/src/main/kotlin/ebi/ac/uk/io/sources/FilesListSource.kt

```diff
@@ -1,13 +1,8 @@
 package ebi.ac.uk.io.sources
 
-import ebi.ac.uk.errors.FileNotFoundException
 import java.io.File
 
 class FilesListSource(private val files: List<File>) : FilesSource {
-    override fun exists(filePath: String) = files.any { it.name == filePath }
-
-    override fun getFile(filePath: String): NfsBioFile {
-        val file = files.firstOrNull { it.name == filePath } ?: throw FileNotFoundException(filePath)
-        return NfsBioFile(file)
-    }
+    override fun getFile(path: String, md5: String?): NfsBioFile? =
+        files.firstOrNull { it.name == path }?.let { NfsBioFile(it) }
```

Comment:
Why are we removing the exception? I think in some cases the file won't exist at all

Reply:
now sources return nullable so is caller responsability to throw it or hanble it.